language/resource-viewer/src/fat_type.rs

```diff
@@ -0,0 +1,172 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+//! Loaded representation for runtime types.
+
+use libra_types::{account_address::AccountAddress, vm_status::StatusCode};
+use move_core_types::{
+    identifier::Identifier,
+    language_storage::{StructTag, TypeTag},
+    value::{MoveStructLayout, MoveTypeLayout},
+};
+use std::convert::TryInto;
+use vm::errors::{PartialVMError, PartialVMResult};
+
+use serde::{Deserialize, Serialize};
+
+/// VM representation of a struct type in Move.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub(crate) struct FatStructType {
+    pub address: AccountAddress,
+    pub module: Identifier,
+    pub name: Identifier,
+    pub is_resource: bool,
+    pub ty_args: Vec<FatType>,
+    pub layout: Vec<FatType>,
```

Comment:
We could, since now resource viewer has total control over `FatType` so it fine to tweak it towards its specific needs. Although the more interesting question is: what do we want with it in the long term?
- Are we going to remove it? If so, what could replace it?
- If we decide to keep it, are we concerned about the information bloat?
  - Imagine you have `struct Foo<T> { x: T, y: T, z: T }`, when you instantiate it, the current representation requires making 4 copies of T, recursively. 
  - Would other clients also want something like this?

Reply:
I think FatType should probably be renamed Type (I know, I know) and used by the viewer for its needs.
So field names and any other info should be there.
That is also in a tool crate so we can make changes as we go